### PR TITLE
fix: true return type for return true always function

### PR DESCRIPTION
**Pull request by kocoten1992**
Hi, I'm not sure what I'm doing, I search for `&return.true.always;` in doc-en and try to replace `bool` with `true`, but what about other language..?

I suppose this is not the right place for this PR? We must have some system to synchronize between languages? Do we have a link for that (I'd close this PR and open new PR).

### CODE REVIEW

**Comment by cmb69** [comment]
> We must have some system to synchronize between languages?

We do have that. The primary changes go into this repository (doc-en), and then translators work on their respective repos (e.g. doc-fr, doc-zh, etc.)

So this PR is at the right place, and basically correct, but since these return types only have been changed as of PHP 8.2.0, we need changelog entries for each function.

**Comment by kocoten1992** [comment]
Hi, did you mean changelog as in https://www.php.net/manual/en/function.file-get-contents.php#refsect1-function.file-get-contents-changelog ?

What is the content should be in changelog ?
| Version  | Description |
| ------------- | ------------- |
| 8.2.0  | return type is `false` now |
| ...  | ...  |

**Comment by kocoten1992** [comment]
@cmb69 is this what you mean? If so it ready for another round of review.

File: reference/array/functions/arsort.xml [hunk 1]
```diff
@@ -48,6 +48,30 @@
...
+      <row>
+       <entry>&Version;</entry>
+       <entry>&Description;</entry>
+      </row>
+     </thead>
+     <tbody>
+      <row>
+       <entry>8.2.0</entry>
+       <entry>
+        return type is &true; now
```
Comment (cmb69) [inline reference/array/functions/arsort.xml:66]:
```suggestion
        The return type is &true; now; previously, it was &bool;.
```

However, given that there are many identical changelog entries, I suggest to define a respective XML entity, similar to:

https://github.com/php/doc-en/blob/e5202c6e4918e1532326e05ae99509ee7c8c036f/language-snippets.ent#L1201-L1207

Then the `<row>` can be replaced with the respective entitiy reference.

**Review by cmb69 — COMMENTED** [review]
Thank you! That is basically what I had I mind, but I think this can be improved (see my comment below).

**Comment by kocoten1992** [comment]
Hi @cmb69 , sorry for the delay, I think this is ready, please take another look.

File: language-snippets.ent [hunk 2]
```diff
@@ -718,6 +718,13 @@ searched for within the <link xmlns="http://docbook.org/ns/docbook" linkend="ini
 
 <!-- Returns -->
 
+<!ENTITY return.type.true '<row xmlns="http://docbook.org/ns/docbook">
+ <entry>8.2.0</entry>
+ <entry>
+  The return type is &true; now; previously, it was &bool;
```
Comment (cmb69) [inline language-snippets.ent:724]: Nit: there should be a period at the end of the sentence:
```suggestion
  The return type is &true; now; previously, it was &bool;.
```

**Review by cmb69 — COMMENTED** [review]
Thank you for the update! I've only now checked the stubs for these changes, and only a few functions have changed from `bool` to `true` (namely `arsort()`, `asort()`, `krsort()`, `ksort()`, `sort()`, `uasort()`, `uksort()` and `usort()`). For all others (except maybe the pthreads function), the stubs still say they return `bool` in PHP 8.2. Maybe the stubs are wrong, but in this case they have to be fixed first, before we can document this. Or maybe some return types are only supposed to happen in PHP 8.3 (haven't checked that)? OTOH, if the stubs are correct, we would need to fix the return type section.

File: language-snippets.ent [hunk 3]
```diff
@@ -718,6 +718,13 @@ searched for within the <link xmlns="http://docbook.org/ns/docbook" linkend="ini
 
 <!-- Returns -->
 
+<!ENTITY return.type.true '<row xmlns="http://docbook.org/ns/docbook">
+ <entry>8.2.0</entry>
+ <entry>
+  The return type is &true; now; previously, it was &bool;.
```
Comment (Girgias) [inline language-snippets.ent:724]:
```suggestion
  The return type is &true; now; previously, it was <type>bool</type>.
```

File: reference/intl/intlcalendar/setrepeatedwalltimeoption.xml [hunk 4]
```diff
@@ -64,6 +64,24 @@
...
+       <entry>&Version;</entry>
+       <entry>&Description;</entry>
+      </row>
+     </thead>
+     <tbody>
+      &return.type.true;
+     </tbody>
+    </tgroup>
+   </informaltable>
+  </para>
```
Comment (Girgias) [inline reference/intl/intlcalendar/setrepeatedwalltimeoption.xml:83]:
```suggestion
  </para>
 </refsect1>
```

File: reference/intl/intlcalendar/setskippedwalltimeoption.xml [hunk 5]
```diff
@@ -78,6 +78,23 @@
...
+       <entry>&Version;</entry>
+       <entry>&Description;</entry>
+      </row>
+     </thead>
+     <tbody>
+      &return.type.true;
+     </tbody>
+    </tgroup>
+   </informaltable>
+  </para>
```
Comment (Girgias) [inline reference/intl/intlcalendar/setskippedwalltimeoption.xml:97]:
```suggestion
  </para>
 </refsect1>
```

**Comment by kocoten1992** [comment]
thank you very much! @Girgias

**Pull request merged**